pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -385,30 +508,22 @@ func (e *DockercfgController) createDockerPullSecret(serviceAccount *api.Service
 	}
 	dockercfgContent, err := json.Marshal(&dockercfg)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 	dockercfgSecret.Data[api.DockerConfigKey] = dockercfgContent
 
 	// Save the secret
 	createdSecret, err := e.client.Secrets(tokenSecret.Namespace).Create(dockercfgSecret)
 	if err != nil {
 		// Clean up the generated token secret if we're not going to use it
-		glog.V(2).Infof("deleting unused token secret %s/%s, error creating dockercfgSecret: %v", tokenSecret.Namespace, tokenSecret.Name, err)
+		glog.V(2).Infof("Deleting unused token secret %s/%s, error creating dockercfgSecret: %v", tokenSecret.Namespace, tokenSecret.Name, err)
 		if deleteErr := e.client.Secrets(tokenSecret.Namespace).Delete(tokenSecret.Name); (deleteErr != nil) && !kapierrors.IsNotFound(deleteErr) {
```

Comment:
Under what conditions is this still necessary?  Since we have the name of the token secret we created, we ought to be able to use it to back the docker pull secret instead of creating another one right?